server/src/main/java/org/eclipse/hono/registration/RegistrationMessageFilter.java

```diff
@@ -64,14 +65,33 @@ public static boolean verify(final ResourceIdentifier linkTarget, final Message
          } else {
              return true;
          }
-    }
+     }
 
-     private static boolean hasValidAction(final Message msg) {
-         String action = MessageHelper.getApplicationProperty(
+     private static String getAction(final Message msg) {
+         return MessageHelper.getApplicationProperty(
                  msg.getApplicationProperties(),
                  RegistrationConstants.APP_PROPERTY_ACTION,
                  String.class);
-         return RegistrationConstants.isValidAction(action);
      }
 
+    private static boolean verifyProperties(final String action, final ResourceIdentifier linkTarget, final Message msg) {
+        if(!RegistrationConstants.ACTION_FIND.equals(action)) {
+            return verifyStandardProperties(linkTarget,msg);
+        }
+        else {
+            final String valueProperty = MessageHelper.getApplicationProperty(msg.getApplicationProperties(),
+                    RegistrationConstants.APP_PROPERTY_VALUE, String.class);
+            if (valueProperty == null) {
+                LOG.trace("message [{}] contains no {} application property", msg.getMessageId(), RegistrationConstants.APP_PROPERTY_VALUE);
+                return false;
+            } else {
```

Comment:
I changed as little as possible for my first hono pull request, and key has not been checked before (but device_id of course).  But you are right, it would be nicer to also check for key in the message filter.